themes/qgrid/source/js/menu.js

```diff
@@ -15,26 +18,229 @@ function toggleVisibility(e) {
 	}
 }
 
-function filter(e) {
+function onSearchChange(e) {
 	e.stopImmediatePropagation();
 	const input = e.target || e.srcElement;
-	const filter = input.value.toLowerCase();
+	const search = input.value.toLowerCase();
+	setSearch(search);
+}
+
+function searchOnLoad() {
+	const active = document.querySelector('.active-topic');
+	const search = getSearch();
+
+	if (active && search) {
+		document.getElementById('search').value = search;
+		currentPageSearch(search);
+		menuItemsSearch(search);
+	}
+}
+
+function searchOnHashChange() {
+	const search = getSearch();
+	document.getElementById('search').value = search;
+
+	if (!search) {
+		document.getElementById('search').focus();
+	}
+	currentPageSearch(search);
+	menuItemsSearch(search);
+}
+
+function currentPageSearch(search) {
+	let scrolled = false;
+
+	document.querySelectorAll('.page-wrapper *').forEach(node => {
+		if (SEARCHEABLE_ELEMENTS.includes(node.nodeName)) {
+			const a = node.querySelectorAll('a');
+			const pageItem = (a[a.length - 1] && a[a.length - 1].textContent) ? a[a.length - 1] : node;
+			const { textContent } = pageItem;
+			const searchWords = splitToWords(search);
+			const firstMatch = (testSearch(textContent, search)) ? searchWords[indexOfSearch(textContent, searchWords)] : '';
+			const searchValue = (textContent.toLowerCase().includes(search)) ? search : firstMatch;
+
+			if (highlightText(pageItem, searchValue) && !scrolled) {
```

Comment:
what is a difference between searchWords and searchValue, does not make sense, please think about other names